src/models/Factor.js

```diff
@@ -318,6 +321,21 @@ function (Okta, Q, factorUtil, Util, Errors, BaseLoginModel) {
 
     pushFactorHasAutoPush: function () {
       return this.settings.get('features.autoPush') && this.get('factorType') === 'push';
+    },
+
+    fixDataForCustomHotp: function(attributes) {
+      // Set vendorname from profile during enroll.
+      if (attributes.factorType === 'custom:hotp' && attributes.profiles) {
+        if (attributes.status === 'ACTIVE') {
```

Comment:
This should be NOT_SETUP because for factors that are not enrolled, we want the name of the first profile